src/compiler/checker.ts

```diff
@@ -21983,8 +21984,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 }
             }
             if (reportErrors) {
-                // Elaborate only if we can find a best matching type in the target union

```

Comment:
I removed this `union` from here because it was misleading - the containing function handles both unions and intersections. However, from what I can tell `reportErrors: true` is only ever passed in for unions - I can revert this if requested